### ISSUE TITLE
Move `flutter/flutter/docs/engine` to `flutter/engine/docs`

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#150869]
... from [`flutter/flutter/docs/engine`](https://github.com/flutter/flutter/tree/master/docs/engine) to [`flutter/engine/docs`](https://github.com/flutter/engine/tree/main/docs).

We would also need to update the links in https://github.com/flutter/flutter/wiki/Supporting-legacy-platforms.

The longer we choose not to do this, the more chance folks bookmark or otherwise expect the files to be in the framework repo, so my two cents is we should just bite the bullet and move them. I can volunteer to do the move and update the links, but I want to make sure both of you are OK with this plan first.

I think moving the docs is preferred for multiple reasons:
- The engine team owns the docs, and we work in the engine repo
- It's good to be able to make atomic changes to the docs and code at the same time
- Being able to cross-reference the docs will become easier within the same repo

Can I get an explicit LGTM?

**Comment by chinmaygarde** [comment]
sgtm, go for it!

**Comment by Piinks** [comment]
SGTM! I don't own these docs, the engine team does. :)
The docs ended up where they are after I called for teams to pick the destination of their docs. At the time no one wanted to move them to flutter/engine and they ended up in flutter/flutter, shame no one proposed otherwise when called to action.

Things to do:
* The [original wiki](https://github.com/flutter/flutter/wiki) pages for all the engine docs points to flutter/flutter/docs locations, so those will need to change.
* All of Flutter's public repos (~20) were updated to point to the new docs locations, and will need to be updated again.
* Docs in flutter/flutter/docs have cross links to these engine pages as well.
